File: C#/7 kyu/0016.cs
// Kata: Is n divisible by (...)?
// Link: https://www.codewars.com/kata/558ee8415872565824000007

// Create a function that checks if the first argument n is divisible by all
// other arguments (return true if no other arguments)


// Example:

// (6,1,3)--> true because 6 is divisible by 1 and 3
// (12,2)--> true because 12 is divisible by 2
// (100,5,4,10,25,20)--> true
// (12,7)--> false because 12 is not divisible by 7
// This kata is following kata:
// http://www.codewars.com/kata/is-n-divisible-by-x-and-y

using System;

namespace Codewars {
  public static class KataChallenge {
    public static bool IsDivisible(params int[] numbers)
    {
      for (int i = 1; i< numbers.Length; ++i)
        if (numbers[0] % numbers[i] != 0) 
          return false;
      return true;
    }
  }
}

File: Python/7 kyu/0016.py
# Kata: Maximum Product
# Link: https://www.codewars.com/kata/5a4138acf28b82aa43000117

# Task
# Given an array of integers , Find the maximum product obtained from
# multiplying 2 adjacent numbers in the array. Note that the array size is at
# least 2 and consists a mixture of positive, negative integers and also zeroes.

# Examples
# - [1, 2, 3] returns 6 because the maximum product is obtained from multiplying
#   2∗3=6
# - [9, 5, 10, 2, 24, -1, -48] returns 50 because the maximum product is
#   obtained from multiplying 5∗10=50
# - [-23, 4, -5, 99, -27, 329, -2, 7, -921] returns -14 because the maximum
#   product is obtained from multiplying −2∗7=−14

import math


def adjacent_element_product(array):
    maximum_product = -math.inf
    for i in range(len(array) - 1):
        maximum_product = max(maximum_product, array[i] * array[i + 1])
    return maximum_product
